.github/workflows/build.yml

```diff
@@ -0,0 +1,54 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+env:
+  MONO_TAG: "6.0.0.334"
+
+jobs:
+
+  build-and-test:
+    name: Build and test
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, windows-latest, macos-latest ]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Install .NET Core 2.1
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 2.1.x
+      - name: Install .NET Core 3.1
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 3.1.x
+      - name: Install .NET Core 5.0
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 5.0.x
+      - name: Clean
+        run: dotnet clean ./Castle.Core.sln --configuration Release && dotnet nuget locals all --clear
```

Comment:
Do we need to "clean" in a CI build? If this is to run it locally, then `dotnet nuget locals all --clear` doesn't sound like something you really want to run, on the CI server won't it also meant you'll blow away the cache of framework packages extending the build time?

Reply:
This is indeed required. Without this `dotnet clean`, the NuGet package restore can sometimes fail. This problem has been documented elsewhere, and &ndash; at least at the time when I prepared this &ndash; the `dotnet clean` appears to be the best-known solution. I can probably locate the issue again that mentions all this.